copy/single.go

```diff
@@ -135,6 +150,7 @@ func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.Unpar
 		ic.compressionFormat = c.options.DestinationCtx.CompressionFormat
 		ic.compressionLevel = c.options.DestinationCtx.CompressionLevel
 	}
+	ic.requireCompressionFormatMatch = requireCompressionFormatMatch
```

Comment:
(Absolutely non-blocking: if it is unconditional like this, the field can be set directly set in the `imageCopier{` constructor above.)